src/types/typechecker.rs

```diff
@@ -1728,6 +1728,28 @@ impl<'a> Inferable<'a> for ast::If<'a> {
     }
 }
 
+impl<'a> Inferable<'a> for ast::Else<'a> {
+    fn infer_impl(&mut self, cache: &mut ModuleCache<'a>) -> TypeResult {
+        // Create new serogate Variable x
+        let x = next_type_variable(cache);
+        //let x = Type::TypeVariable(id);
+        // Construct a Maybe x Type
+        //let maybe_typeinfo_id = cache.type_infos.get(cache.maybe_type.0);
```

Comment:
```suggestion
```
No need to repeat in comments what is already done on the code in the next lines.

Comment-wise, I usually only leave comments to explain  _why_ I'm doing something rather than _what_ I'm doing. For example, if there's a specific edge case that prevents the code from being simpler I'll write that down so I don't try to simplify the code later and break it.

Other than that though, type checking code looks good